src/widgets/DashboardPage.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2018
+ */
+
+import React, { useState, useEffect } from 'react';
+import { View, StyleSheet } from 'react-native';
+import PropTypes from 'prop-types';
+import globalStyles from '../globalStyles';
+import { testData } from '../database/utilities/ReportDemoData';
+import { ReportSideBar } from './ReportSideBar';
+import { ReportChart } from './ReportChart';
+import { createOrUpdateRecord } from '../sync/incomingSyncUtils';
+
+export const DashboardPage = ({ database, settings }) => {
+  const [reports, setReports] = useState(null);
+  const [selectedItemIndex, setSelectedItemIndex] = useState(0);
+
+  // Used for adding data within ./ReportDemoData to the database for
+  // testing while syncing from mobile is not functional.
+  database.write(() =>
+    testData.forEach(report => {
+      const recordCopy = { ...report };
+      recordCopy.data = JSON.stringify(report.data);
+      createOrUpdateRecord(database, settings, 'Report', recordCopy);
+    })
+  );
+
+  useEffect(() => {
+    setReports(
+      database
+        .objects('Report')
+        .snapshot()
+        .map((report, index) => ({
+          id: report.reportID,
+          index,
+          title: report.title,
+          type: report.type,
+          date: report.date,
+          data: JSON.parse(report.data),
+        }))
+    );
+  });
```

Comment:
`useEffect` need a dependencies list to manage when it should be called. At the moment, this will run every time the component renders, and will call `setReports` which will re-render the component.. causing an infinite loop!

needs to be `useEffect(() => { ... }, [])` in order to only be set when the component mounts.

Reply:
Nice, thanks! I added `useEffect(() => { ... }, [reports])`. That way, I _think_ that `useEffect` will only re-run if `reports` changes. Is that correct?